### PR TITLE
Devise accounts photo scaffold

### PR DESCRIPTION
**Pull request by Testell**

### CODE REVIEW

**Review by dsimp — APPROVED** [review]

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,3 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
```
Comment (JoseMDO) [inline app/controllers/application_controller.rb:2]: Nice addition of authentication!

File: app/controllers/photos_controller.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    # Use callbacks to share common setup or constraints between actions.
+    def set_photo
+      @photo = Photo.find(params[:id])
+    end
+
+    # Only allow a list of trusted parameters through.
+    def photo_params
+      params.require(:photo).permit(:image, :comments_count, :likes_count, :caption, :owner_id)
+    end
+end
```
Comment (JoseMDO) [inline app/controllers/photos_controller.rb:70]: Controller looks great!

File: app/models/photo.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+# Indexes
+#
+#  index_photos_on_owner_id  (owner_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (owner_id => users.id)
+#
+class Photo < ApplicationRecord
+  belongs_to :owner, class_name: "User"
```
Comment (JoseMDO) [inline app/models/photo.rb:23]: Association is good

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+#  index_users_on_reset_password_token  (reset_password_token) UNIQUE
+#  index_users_on_username              (username) UNIQUE
+#
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  has_many :own_photos, class_name: "Photo", foreign_key: "owner_id"
```
Comment (JoseMDO) [inline app/models/user.rb:30]: Nice

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -6,11 +6,19 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
+    <%= render partial: "shared/cdn_assets" %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:9]: Great use of partials :)

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -6,11 +6,19 @@
...
     <%= javascript_importmap_tags %>
   </head>
 
   <body>
-    <%= yield %>
+
+  <%= render partial: "shared/navbar"%>
+
+   <div class="container mt-3">
+      <%= render partial: "shared/flash_messages" %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:20]: Same as above

File: config/routes.rb [hunk 7]
```diff
@@ -1,6 +1,8 @@
 Rails.application.routes.draw do
```
Comment (JoseMDO) [inline config/routes.rb:1]: Routes look good

**Review by JoseMDO — APPROVED** [review]
Great job Tyler!

**Pull request merged**